client/packages/common/src/ui/layout/tables/material-react-table/useNonPaginatedMaterialTable.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+  MRT_RowData,
+  MRT_TableOptions,
+  useMaterialReactTable,
+} from 'material-react-table';
+import {
+  CheckboxCheckedIcon,
+  CheckboxEmptyIcon,
+  CheckboxIndeterminateIcon,
+} from '@common/icons';
+
+interface NonPaginatedTableConfig<T extends MRT_RowData>
+  extends MRT_TableOptions<T> {
+  onRowClick?: (row: T) => void;
+  isLoading: boolean;
+}
+
+export const useBaseMaterialTable = <T extends MRT_RowData>({
+  isLoading,
+  onRowClick,
+  ...tableOptions
+}: NonPaginatedTableConfig<T>) => {
+  const table = useMaterialReactTable<T>({
+    enablePagination: false,
+    enableRowVirtualization: true,
+    enableColumnResizing: true,
+    enableColumnPinning: true,
+    enableColumnOrdering: true,
+    enableColumnDragging: false,
+    enableRowSelection: true,
+
+    // Disable bottom footer - use OMS custom action footer instead
+    enableBottomToolbar: false,
+
+    initialState: {
+      density: 'compact',
+      columnPinning: { left: ['mrt-row-select'] },
+      ...tableOptions.initialState,
+    },
+    state: {
+      showProgressBars: isLoading,
+    },
+
+    // Styling
+    muiTablePaperProps: {
+      sx: { width: '100%', display: 'flex', flexDirection: 'column' },
+    },
+    muiTableHeadCellProps: {
+      sx: {
+        fontWeight: 600,
+        lineHeight: 1.2,
+        verticalAlign: 'bottom',
+        justifyContent: 'space-between',
+        '& .Mui-TableHeadCell-Content svg': {
+          fontSize: '2em',
+          marginLeft: 0,
+        },
+        // Allow date range filters to wrap if column is too narrow
+        '& .MuiCollapse-wrapperInner > div': {
+          display: 'flex',
+          flexWrap: 'wrap',
+          // Date picker should never need to be wider than 170px
+          '& .MuiPickersTextField-root': { width: '170px' },
+        },
+      },
+    },
+    muiTableBodyCellProps: {
+      sx: { fontSize: '14px', fontWeight: 400 },
+    },
+
+    muiTopToolbarProps: {
+      sx: { height: '60px' }, // Prevent slight jump when selecting rows
+    },
+
+    muiSelectAllCheckboxProps: {
+      color: 'outline',
+      size: 'small',
+      icon: <CheckboxEmptyIcon />,
+      checkedIcon: <CheckboxCheckedIcon />,
+      indeterminateIcon: <CheckboxIndeterminateIcon />,
+    },
+    muiSelectCheckboxProps: {
+      color: 'outline',
+      size: 'small',
+      icon: <CheckboxEmptyIcon />,
+      checkedIcon: <CheckboxCheckedIcon />,
+      indeterminateIcon: <CheckboxIndeterminateIcon />,
+    },
+    muiToolbarAlertBannerProps: {
+      sx: { backgroundColor: 'unset' },
+    },
+
+    muiTableBodyRowProps: ({ row }) => ({
+      onClick: () => {
+        if (onRowClick) onRowClick(row.original);
+      },
+      sx: {
+        '& td': { borderBottom: '1px solid rgba(224, 224, 224, 1)' },
+      },
+    }),
+
+    ...tableOptions,
+  });
+  return table;
+};
+
+// Export a renamed version for clarity
+export const useNonPaginatedMaterialTable = useBaseMaterialTable;
```

Comment:
Oh I see, the non-paginated version is just the base version renamed. That relies on the non-paginated version having no settings that are not also in the paginated version, which I'm not totally sure is the case (`enableRowVirtualization` for one). I guess you can just "unset" them in the paginated version, but I think it would be slighly cleaner to have the two variants both use the base and only put *common* settings in the base one (especially if we move "Base" to its own file, then it would be really clear to see which properties are common, and what are the unique properties for each variant)

Reply:
Yep fair call, especially with the virtualisation, hadn't noticed that, will update now 🙏

Reply:
[8365919](https://github.com/msupply-foundation/open-msupply/pull/9176/commits/8365919f6b11ccea9a51178568f0b4959f4143ab)